External/Plugins/LintingHelper/LintingCache.cs

```diff
@@ -17,6 +17,14 @@ public LintingCache()
             results = new Dictionary<string, List<LintingResult>>();
         }
 
+        /// <summary>
+        /// Returns all cached results of all files that are still in the cache
+        /// </summary>
+        public List<LintingResult> GetAllResults()
+        {
+            return new List<LintingResult>(results.Values.SelectMany(i => i));
```

Comment:
I think that you should use ```results.Values.SelectMany(i => i).toList();``` instead of ```new List<LintingResult>(results.Values.SelectMany(i => i));```